src/components/application_manager/src/commands/hmi/basic_communication_on_awake_sdl.cc

```diff
@@ -29,3 +29,45 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#include "application_manager/commands/hmi/basic_communication_on_awake_sdl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/resumption/resume_ctrl.h"
+
+namespace application_manager {
+
+namespace commands {
+
+OnAwakeSDLNotification::OnAwakeSDLNotification(
+    const MessageSharedPtr& message, ApplicationManager& application_manager)
+    : NotificationFromHMI(message, application_manager) {}
+
+OnAwakeSDLNotification::~OnAwakeSDLNotification() {}
+
+void OnAwakeSDLNotification::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (application_manager_.resume_controller().is_suspended()) {
+    {
+      application_manager_.resume_controller().set_is_suspended(false);
+      DataAccessor<ApplicationSet> accessor =
+          application_manager_.applications();
+      ApplicationSetIt it = accessor.GetData().begin();
+      ApplicationSetIt itEnd = accessor.GetData().end();
+      for (; it != itEnd; ++it) {
+        if ((*it).get()) {
```

Comment:
@AKalinich-Luxoft please move `it` to local variable to avoid tones of (*)-> ...